### PR TITLE
add repo settings

### LINKED ISSUE #855: Clean up labels
@ewelinawilkosz @timja 

I want to clean up some of our labels

<details><summary>Full list</summary>

```
good first issue
Status - in progress
Action - awaiting feed-back
Action - needs grooming
Prio 1 - must have
Prio 2 - should have
Prio 3 - could have
Prio 4 - won’t have
Size 0 - Briefing
Size 1 - small
Size 2 - medium
Size 3 - large
Size 4 - too big
Status - duplicate
Status - workable
Status - up Next
hackathon
beta-release
help wanted
question
wontfix
invalid
duplicate
enhancement
bug
Work in progress
documentation
do not merge
pending PR
hacktoberfest
needs-plugin-issue
feature
fix
chore
test
pinned
plugin-compatibility
stale
reverted
```

</details>

Here's what I suggest

<details><summary>Remove</summary>
<p>

label : reason
```
Status - in progress : unused
Action - awaiting feed-back : unused
Action - needs grooming : unused
Prio 1 - must have : unused
Prio 2 - should have : unused
Prio 3 - could have : unused
Prio 4 - won’t have : unused
Size 0 - Briefing : unused
Size 1 - small : unused
Size 2 - medium : unused
Size 3 - large : unused
Size 4 - too big : unused
Status - duplicate : unused
Status - workable : unused
Status - up Next : unused
beta-release : unused
Work in progress : replaced by WIP[bot]
do not merge : replaced by WIP[bot]
pending PR : debatable just as easily marked by WIP or 🚧
needs-plugin-issue - replaced by plugin-compatibility
invalid : replaced by stale[bot]
```

</p>
</details>

<details><summary>Stay</summary>
<p>

label : reason
```
good first issue : used by github for finding new contributors
help wanted : used by github for finding new contributors
hackathon : useful for hackathon events
question : signal value
wontfix : useful to find things we wontfix
duplicate : signal value
enhancement : enhancement we want to get to (exempted from stale[bot])
bug : signal value
documentation : release drafter
hacktoberfest : event
feature : release drafter
fix : release drafter
chore : release drafter
test : release drafter
pinned : exempted from stale[bot]
plugin-compatibility : used by issue template
stale : stale[bot]
reverted : release drafter
```

</p>
</details>

Some that require extra explanation.

`invalid` would be the same as stale since stale[bot] will clean up if a user doesn't reply within 7 days if we mark it stale.

`Work in progress`/`do not merge` add WIP or 🚧 to PR title.
I will update the PR template to make this clear.

I suggest we use 🚧 as a convention as we do like emojis :trollface: 

Perhaps we should replace `enhancement` with `feature` to avoid confusion  
(I prefer the word feature over enhancement)

I would like to use [probot/settings](https://github.com/probot/settings) to achieve this cleanup :sweat_smile:

### PR DESCRIPTION
**Pull request by casz**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [x] Link to relevant GitHub issues or pull requests

- [ ] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

fixes #855

Just doing the grunt work to prepare for configuration as code of our github repo settings :sweat_smile:

Please continue the discussion in #855 about labels to cleanup and than they will be carried out here. :sweat:

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+labels:
+  - name: good first issue
+    color: 7057ff
+  - name: Status - in progress
+    color: ededed
+  - name: Action - awaiting feed-back
+    color: 6eb82c
+  - name: Action - needs grooming
+    color: 009800
+  - name: Prio 1 - must have
```
Comment (timja) [inline .github/settings.yml]: do we need all these labels?

Comment (casz) [reply .github/settings.yml]: See #855 :laughing:

Comment (timja) [reply .github/settings.yml]: just saw

**Comment by casz** [comment]
@daniel-beck could you approve my request for "settings" to jenkinsci org? https://github.com/probot/settings

**Comment by casz** [comment]
This PR is currently blocked by requested app approval. I have a request for "[settings](https://github.com/probot/settings)" waiting to be approved for configuration-as-code-plugin repo
@daniel-beck I think a `@jenkinsci/admin` group would make sense so I can poke more than one individual 😓 or is there a better channel? should I use the mailing list or should I use Jira. I find it confusing that we live outside of GitHub.

**Comment by casz** [comment]
Still blocked.
Opened https://issues.jenkins-ci.org/browse/INFRA-2103
and poked the mailing list https://groups.google.com/forum/#!topic/jenkinsci-dev/QNjrd1zVXZY

**Comment by timja** [comment]
@casz wasn't this approved?

**Comment by casz** [comment]
The only blocking thing is automated releases since I constantly have to remove WIP and re-add WIP branch protection.

If GitHub would solve turning a PR into a draft PR we wouldn't need the WIP branch protection.

File: .github/settings.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  topics: 'configuration-as-code, jenkins, hacktoberfest, jenkins-configuration, jcasc'
+  private: false
+  default_branch: master
+  has_issues: true
+  has_wiki: false
+  allow_merge_commit: false
+  allow_squash_merge: true
+  allow_rebase_merge: false
+
+labels:
```
Comment (oleg-nenashev) [inline .github/settings.yml]: Could we move it to https://github.com/jenkinsci/.github ? IIUC the app supports Probot, and having some default labels would not hurt if others from Web UI do not get removed

Comment (casz) [reply .github/settings.yml]: https://github.com/probot/settings/issues/107
Currently, these cannot be moved to .github as they would delete other labels that are not inside settings.yml

Comment (oleg-nenashev) [reply .github/settings.yml]:
```suggestion
#TODO(oleg-nenashev): Consider moving to https://github.com/jenkinsci/.github once https://github.com/probot/settings/issues/107 is fixed
labels:
```

**Comment by casz** [comment]
WIP branch protection is only somewhat of a blocker since I would like repo settings to include branch protection but I think cleaning up labels will be helpful.

**Comment by timja** [comment]
any reason not to 🚢 this @casz?

**Comment by casz** [comment]
:ship: :100:

**Comment by casz** [comment]
lol, the approval for the app was removed. @oleg-nenashev could you fix? :|

**Comment by casz** [comment]
I see that they have been successfully applied.

So now we have 22 labels 🙌

**Pull request merged**